### DISCUSSION TITLE
WPF - Cursor gets set to "None" locally

### DISCUSSION THREAD

**Discussion by DJmRek** [cefsharp/CefSharp#4035] [General]
I recently updated a wpf app from CefSharp 91 to 98. I'm facing a weird issue where the Cursor property on the ChromiumWebBrowser Control gets set to "None" which hides the cursor. I searched our codebase as well as the cefsharp codebase but didn't find any clue what can cause this nor any trace of any cursor setting action.
Strangly enough the MinimalExample works. When I alter the value at runtime via snoop it flicks back to "None" as soon as i hover the mouse over the control.

Any ideas what can cause this?
Did I miss any crucial changes between v91 and v98?

Thanks in advance for any feedback.

**Comment by amaitland** [comment]
Are you using WinForms or WPF?

**Reply by amaitland** [reply]
Also are there any errors in the log? 

https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file

**Reply by DJmRek** [reply]
Sorry for the missing information. I use WPF. I dug a bit deeper today and found, that in fact its not the cursor affected by WPF but the cursor within the browser. I figured that in the minimal example the cursor property of the WPF component is set to "None" also. Which makes sense since the browser renders the cursor on its own.

I narrowed it down to the point, that it is cause by the used cursor in css. The website uses the "grab"/"-webkit-grab" cursor mainly. This cursor doesn't work in the minial-example v98 and v99. Strangly enough other cursors like "pointer" work and the grab cursor also works in Chrome 99.0.4844.51.
Maybe this is a problem of chromium?

**Reply by DJmRek** [reply]
@amaitland i will try a few of the earlier versions to narrow it down. v91 is known working.
Can I open a PR  for a small MinimalExample improvement? (I added a Button to open DevTools in MinimalExample.Wpf)

**Reply by DJmRek** [reply]
I narrowed it down:
v94-v96 are working
v97/v98/v99 are not working

**Reply by amaitland** [reply]
> I use WPF

This is an upstream bug

https://bitbucket.org/chromiumembedded/cef/issues/3270/windows-osr-mouse-cursors-not-changing-as

> I narrowed it down:
v94-v96 are working
v97/v98/v99 are not working

If you can add the info to the bitbucket issue It might help narrow down what changed in Chromium. 

I've proposed a workaround for the pan cursor in https://github.com/cefsharp/CefSharp/issues/4021#issuecomment-1054895227

You can likely expand that to handle your case.

> Can I open a PR for a small MinimalExample improvement? (I added a Button to open DevTools in MinimalExample.Wpf)

Sure 👍

**Reply by DJmRek** [reply]
Thx for the feedback. I will add the info at the chromium upstream bug. I can confirm that it only happens when `--off-screen-rendering-enabled` flag is used as you mentioned in #4021. Seems to affect quit a bunch of the cursor options.

One question tho, is the chromium instance always started with this falg even if I use WinForms/WPF?

**Reply by amaitland** [reply]
> One question tho, is the chromium instance always started with this falg even if I use WinForms/WPF?

The WinForms and Wpf.HwndHost implentations will be unaffected.

**Reply by amaitland** [reply]
I've implemented a workaround in the `WPF` implementation which is now available in the [100.0.120-pre](https://github.com/cefsharp/CefSharp/issues/4053#issuecomment-1088135006) release. 

Anyone interested please test and report back on https://github.com/cefsharp/CefSharp/issues/4021
